contracts/proxy/utils/Initializable.sol

```diff
@@ -91,7 +77,15 @@ abstract contract Initializable {
         _;
     }
 
-    function _isConstructor() private view returns (bool) {
-        return !Address.isContract(address(this));
+    function _setInitializeStep(uint8 version) private returns (bool) {
+        if (_initializing) {
+            require(version == 1 && !Address.isContract(address(this)), "Initializable: contract is already initialized");
```

Comment:
Isn't this what we want? We should also add a comment explaining it. In fact we already have a comment but it's being deleted above.

```suggestion
            require(version > 1 || !Address.isContract(address(this)), "Initializable: contract is already initialized");
```

Reply:
IMO, this should not work outside a constructor.

The current version is 
```
        require(_initializing ? _isConstructor() : !_initialized, "Initializable: contract is already initialized");

        bool isTopLevelCall = !_initializing;
        if (isTopLevelCall) {
            _initializing = true;
            _initialized = true;
        }

        _;

        if (isTopLevelCall) {
            _initializing = false;
        }
```

So right now, an `initializer()` function will set both `_initializing` and `_initialized`
This means that is another such call is nested, we will require that we are in the constructor.

Today, initializer nesting is only allowed in constructor, which is what `&&` does. Using `||` would be a breaking change

Reply:
Ok, sorry, I understand now. Can you add back the comment documenting this behavior?